### PR TITLE
backport: 'trivial' pr11

### PR DESCRIPTION
**Pull request by PastaPastaPasta**

### CODE REVIEW

File: doc/productivity.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ Table of Contents
    * [Multiple working directories with `git worktrees`](#multiple-working-directories-with-git-worktrees)
    * [Interactive "dummy rebases" for fixups and execs with `git merge-base`](#interactive-dummy-rebases-for-fixups-and-execs-with-git-merge-base)
 * [Writing code](#writing-code)
-   * [Format C/C++/Protobuf diffs with `clang-format-diff.py`](#format-ccprotobuf-diffs-with-clang-format-diffpy)
+   * [Format C/C++ diffs with `clang-format-diff.py`](#format-cc-diffs-with-clang-format-diffpy)
```
Comment (PastaPastaPasta) [inline doc/productivity.md:15]: drop 19058

File: src/crc32c/.appveyor.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# Build matrix / environment variables are explained on:
```
Comment (PastaPastaPasta) [inline src/crc32c/.appveyor.yml:1]: maybe drop d386a4db0c0e060564b3864b9d7da82f79171d47

File: src/checkqueue.h [hunk 3]
```diff
@@ -92,8 +92,7 @@ class CCheckQueue
                         nTotal--;
                         bool fRet = fAllOk;
                         // reset the status for new work later
-                        if (fMaster)
-                            fAllOk = true;
+                        fAllOk = true;
```
Comment (PastaPastaPasta) [inline src/checkqueue.h:95]: need to drop 17971

File: test/functional/mempool_accept.py [hunk 4]
```diff
@@ -248,6 +248,12 @@ def run_test(self):
             rawtxs=[bytes_to_hex_str(tx.serialize())],
         )
         tx.deserialize(BytesIO(hex_str_to_bytes(raw_tx_reference)))
+        tx.vin[0].scriptSig = CScript([b'a' * 1648]) # Some too large scriptSig (>1650 bytes)
+        self.check_mempool_result(
+            result_expected=[{'txid': tx.rehash(), 'allowed': False, 'reject-reason': 'scriptsig-size'}],
```
Comment (UdjinM6) [inline test/functional/mempool_accept.py:253]: 17532:
```suggestion
            result_expected=[{'txid': tx.rehash(), 'allowed': False, 'reject-reason': '64: scriptsig-size'}],
```

File: src/test/transaction_tests.cpp [hunk 5]
```diff
@@ -351,6 +351,29 @@ BOOST_AUTO_TEST_CASE(test_IsStandard)
...
+    reason.clear();
+    BOOST_CHECK(!IsStandardTx(CTransaction(t), reason));
+    BOOST_CHECK_EQUAL(reason, "version");
+
+    t.nVersion = 0;
+    reason.clear();
+    BOOST_CHECK(!IsStandardTx(CTransaction(t), reason));
+    BOOST_CHECK_EQUAL(reason, "version");
+
+    t.nVersion = 3;
```
Comment (UdjinM6) [inline src/test/transaction_tests.cpp:365]: 17555:
```suggestion
    t.nVersion = 4;
```

File: src/test/transaction_tests.cpp [hunk 6]
```diff
@@ -351,6 +351,29 @@ BOOST_AUTO_TEST_CASE(test_IsStandard)
...
+    BOOST_CHECK(!IsStandardTx(CTransaction(t), reason));
+    BOOST_CHECK_EQUAL(reason, "version");
+
+    // Allowed nVersion
+    t.nVersion = 1;
+    BOOST_CHECK(IsStandardTx(CTransaction(t), reason));
+
+    t.nVersion = 2;
+    BOOST_CHECK(IsStandardTx(CTransaction(t), reason));
+
```
Comment (UdjinM6) [inline src/test/transaction_tests.cpp:376]: 17555:
```suggestion

    t.nVersion = 3;
    BOOST_CHECK(IsStandardTx(CTransaction(t), reason));
```

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

**Comment by PastaPastaPasta** [comment]
Let's see how CI likes that. Will squash once it's happy

**Comment by PastaPastaPasta** [comment]
Squashed and fped

**Review by UdjinM6 — APPROVED** [review]
utACK

**Pull request merged**